robots/commenter/main.go

```diff
@@ -209,13 +238,35 @@ func makeCommenter(comment string, useTemplate bool) func(meta) (string, error)
 	}
 }
 
-func run(c client, query, sort string, asc, random bool, commenter func(meta) (string, error), ceiling int) error {
-	log.Printf("Searching: %s", query)
-	issues, err := c.FindIssues(query, sort, asc)
-	if err != nil {
-		return fmt.Errorf("search failed: %v", err)
+func run(ctx context.Context, c client, gh *github2.Client, queries []string, sort string, asc, random bool, commenter func(meta) (string, error), ceiling int, labelsUpdated time.Duration) error {
+	var issues []github.Issue
+	issueAlreadyHandled := map[int]bool{}
+	for _, query := range queries {
+		log.Printf("Searching: %s", query)
+		foundIssues, err := c.FindIssues(query, sort, asc)
+		if err != nil {
+			return fmt.Errorf("search failed: %v", err)
+		}
+		for _, foundIssue := range foundIssues {
+			if issueAlreadyHandled[foundIssue.ID] {
+				continue
+			}
+			issueAlreadyHandled[foundIssue.ID] = true
+
```

Comment:
what do you think about separating the issue query and the labels filtering in two for loops?

Reply:
If you mean to put the labels filtering issue out of the query loop, than this is not possible as we need the separate labels of each query